packages/python/accern_xyme/types.py

```diff
@@ -21,7 +21,8 @@
     "nodes": List['NodeDef'],
     "queue_mng": Optional[str],
     "uri_prefix": Optional[URIPrefix],
-})
+    "state": str,
+}, total=False)
```

Comment:
uri_prefix may or maynot be there

Reply:
then your get_uri_prefix function and get_state_type/uri would be wrong. also, if it's not there then it would be Optional not total -- but since we're reading dagdef from the server it will always be there -- since monitor will return always the latest version even if it is the old version on disk

Reply:
i.e., there is no way monitor could respond  with any of the fields being unset (except input/output/queue) or None. uri_prefix and state_uri are guaranteed to exist and have meaningful values

Reply:
except if you want to be more permissive for set_dag in which case you should probably use a duplicate type with total=False?

Reply:
(or just Dict[str, Any])